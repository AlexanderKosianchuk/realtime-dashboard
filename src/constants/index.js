import { keyMirror } from 'modules/helpers';

export const AlertActionTypes = keyMirror({
  SHOW_ALERT: undefined,
  HIDE_ALERT: undefined,
});

export const DataReceiverActionTypes = keyMirror({
  DATA_RECEIVER_CONNECT: undefined,
  DATA_RECEIVER_DISCONNECT: undefined,
  DATA_RECEIVER_SET_STATUS: undefined,
  DATA_RECEIVER_SET_DATA: undefined,
});

export const CONNECTION_STATUS = {
  IDLE: 'idle',
  CONNECTION_ATTEMPT: 'connectionAttempt',
  CONNECTED: 'connected',
  DISCONNECTED: 'disconnected',
  ERROR: 'error',
};

export const SOCKET_EVENTS = {
  DATA: 'data',
  ERROR: 'error',
};

export const POINTS_MAX_COUNT = 50;

export const SERVER_URL = process.env.URL || 'http://localhost:4000';
