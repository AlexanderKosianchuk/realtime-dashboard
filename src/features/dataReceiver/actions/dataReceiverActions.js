// @flow
import { createActions } from 'redux-actions';
import { DataReceiverActionTypes } from 'constants/index';

export const {
  dataReceiverConnect,
  dataReceiverDisconnect,
  dataReceiverSetStatus,
  dataReceiverSetData
} = createActions({
  [DataReceiverActionTypes.DATA_RECEIVER_CONNECT]: () => {},
  [DataReceiverActionTypes.DATA_RECEIVER_DISCONNECT]: () => {},
  [DataReceiverActionTypes.DATA_RECEIVER_SET_STATUS]: status => ({ status }),
  [DataReceiverActionTypes.DATA_RECEIVER_SET_DATA]: data => ({ data }),
});
