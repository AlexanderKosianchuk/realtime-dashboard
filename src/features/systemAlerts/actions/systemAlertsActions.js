import uid from 'nanoid';
import { createActions } from 'redux-actions';
import { AlertActionTypes } from 'constants/index';

export const { hideAlert, showAlert } = createActions({
  [AlertActionTypes.HIDE_ALERT]: (id: string) => ({ id }),
  [AlertActionTypes.SHOW_ALERT]: (message: string, options: Object) => {
    const timeout = options.variant === 'danger' ? 0 : 2;

    return {
      id: options.id || uid(),
      icon: options.icon,
      message,
      position: options.position || 'bottom-right',
      variant: options.variant || 'dark',
      timeout: typeof options.timeout === 'number' ? options.timeout : timeout,
    };
  },
});
