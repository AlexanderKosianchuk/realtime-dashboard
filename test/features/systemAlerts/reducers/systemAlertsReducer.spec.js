import reducer from 'features/systemAlerts/reducers/systemAlertsReducer';
import { hideAlert, showAlert } from 'features/systemAlerts/actions/systemAlertsActions';

import { AlertActionTypes } from 'constants/index';

describe('systemAlerts systemAlertsReducer', () => {
  let systemAlerts = reducer.systemAlerts(undefined, {});

  it('should return the initial state', () => {
    expect(reducer.systemAlerts(systemAlerts, {})).toMatchSnapshot();
  });

  it(`should handle ${AlertActionTypes.SHOW_ALERT}`, () => {
    systemAlerts = reducer.systemAlerts(systemAlerts, showAlert('HELLO', { id: 'test', type: 'success' }));
    expect(systemAlerts).toMatchSnapshot();
  });

  it(`should handle ${AlertActionTypes.HIDE_ALERT}`, () => {
    systemAlerts = reducer.systemAlerts(systemAlerts, hideAlert('test'));
    expect(systemAlerts).toMatchSnapshot();
  });
});
