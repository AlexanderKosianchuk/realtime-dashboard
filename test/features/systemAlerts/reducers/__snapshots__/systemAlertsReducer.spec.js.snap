// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`systemAlerts systemAlertsReducer should handle HIDE_ALERT 1`] = `
Object {
  "alerts": Array [],
}
`;

exports[`systemAlerts systemAlertsReducer should handle SHOW_ALERT 1`] = `
Object {
  "alerts": Array [
    Object {
      "icon": undefined,
      "id": "test",
      "message": "HELLO",
      "position": "bottom-right",
      "timeout": 2,
      "variant": "dark",
    },
  ],
}
`;

exports[`systemAlerts systemAlertsReducer should return the initial state 1`] = `
Object {
  "alerts": Array [],
}
`;
